piet/src/text.rs

```diff
@@ -41,10 +67,25 @@ pub trait TextLayoutBuilder {
 /// - If the text position is not at a code point or grapheme boundary, undesirable behavior may
 /// occur.
 ///
-pub trait TextLayout {
+pub trait TextLayout: Clone {
     /// Measure the advance width of the text.
     fn width(&self) -> f64;
 
+    /// Used for changing the width of a text layout. Given a width, returns a [`TextLayout`]
+    /// struct with recalculated lines and line metrics.
+    // TODO: Should this take &self or self? I'd say maybe; perhaps the old struct should be kept around
+    // to help with caching.
+    fn update_width(&self, new_width: f64) -> Self;
+
+    /// Given a line number, return a reference to that line's underlying string.
+    fn line_text(&self, line_number: usize) -> Option<&str>;
+
+    /// Given a line number, return a reference to that line's metrics.
+    fn line_metric(&self, line_number: usize) -> Option<&LineMetric>;
```

Comment:
if `LineMetric` doesn't contain anything expensive, this should return it by value, not by reference.